rero_ils/modules/notifications/dispatcher.py

```diff
@@ -157,19 +160,32 @@ def not_yet_implemented(*args):
                 if end_date:
                     documents_data['end_date'] = end_date
                 # Add item to document
-                item = loan.get('item')
-                if item:
+                item_data = loan.get('item')
+                if item_data:
+                    if notification_type in [
+                        Notification.BOOKING_NOTIFICATION_TYPE
+                    ]:
+                        # get item from the checkin loan
+                        item = Item.get_record_by_pid(
+                            item_data.get('pid'))
+                        # get the requested loan
+                        request_loan = item.get_first_loan_by_state(
```

Comment:
Use case to check:

You need to check the case when there is an automatic validation i.e. item is checked-in and system will automatically `validate` the next `pending` request. In this case, the next `request` has the state `ITEM_AT_DESK` and not `PENDING`

If the code of notification creation is executed before the `automatic` validation, then there is no problem